benches/benches/bevy_ecs/change_detection.rs

```diff
@@ -69,11 +82,14 @@ fn all_added_detection_generic<T: Component + Default>(group: &mut BenchGroup, e
     group.bench_function(
         format!("{}_entities_{}", entity_count, std::any::type_name::<T>()),
         |bencher| {
-            bencher.iter_batched(
-                || setup::<T>(entity_count),
-                |mut world| {
+            bencher.iter_batched_ref(
+                || {
+                    let mut world = setup::<T>(entity_count);
+                    let query = world.query_filtered::<Entity, Added<T>>();
```

Comment:
We should have a comment here, explaining why we're doing this outside of the loop.